src/GraphQL/Types/EnumerationGraphType.cs

```diff
@@ -112,16 +112,22 @@ public class EnumValues : IEnumerable<EnumValueDefinition>
 
         public void Add(EnumValueDefinition value) => _values.Add(value ?? throw new ArgumentNullException(nameof(value)));
 
-        public IEnumerator<EnumValueDefinition> GetEnumerator() => _values.GetEnumerator();
+        public IEnumerator<EnumValueDefinition> GetEnumerator() => _values.Where(x => x.Visible).GetEnumerator();
 
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
     }
 
     public class EnumValueDefinition : MetadataProvider
     {
+        public EnumValueDefinition()
+        {
+            Visible = true;
+        }
+
         public string Name { get; set; }
         public string Description { get; set; }
         public string DeprecationReason { get; set; }
         public object Value { get; set; }
+        public bool Visible { get; set; }
```

Comment:
No need for ctor, just
```c#
public bool Visible { get; set; } = true;
```

Reply:
Wasn't sure 'bout the convention, thanks :3